hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManager.java

```diff
@@ -517,7 +547,7 @@ private boolean doConnectToCandidateCluster(CandidateClusterContext context, boo
             logger.warning("Stopped trying on the cluster: " + context.getClusterName()
                     + " reason: " + e.getMessage());
         }
-
+        connectionProcessListener.clusterConnectionFailed(context.getClusterName());
```

Comment:
Why don't we fire these events in the executor as well?

Reply:
we are not under the mutex here